File: Scripts/results/generated_code_results/RN_gpt/Social Networking/SocialApp/HomepageScreen.js
import React from 'react';
import { View, Text, TextInput, Image, Button, TouchableOpacity, StyleSheet, ScrollView } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

const HomepageScreen = ({ navigation }) => {
  return (
    <ScrollView style={styles.container}>
      <View style={styles.header}>
        <TouchableOpacity style={styles.menuButton}>
          <Icon name="menu" size={30} />
        </TouchableOpacity>
        <Text style={styles.title}>Forging New Bonds and Friendships</Text>
        <View style={styles.headerIcons}>
          <TouchableOpacity onPress={() => {}}>
            <Icon name="notifications-outline" size={30} />
          </TouchableOpacity>
          <TouchableOpacity onPress={() => {}}>
            <Icon name="settings-outline" size={30} />
          </TouchableOpacity>
        </View>
      </View>

      <ScrollView horizontal style={styles.stories}>
        <View style={styles.story}>
          <Image source={require('../assets/snack-icon.png')} style={styles.storyImage} />
          <Text>My Story</Text>
          <View style={styles.addStoryIcon}>
            <Icon name="add-circle-outline" size={20} />
          </View>
        </View>

        {['Hasley', 'Javier', 'Natalie'].map(name => (
          <View key={name} style={styles.story}>
            <Image source={require('../assets/snack-icon.png')} style={styles.storyImage} />
            <Text>{name}</Text>
          </View>
        ))}
      </ScrollView>

      <View style={styles.searchBar}>
        <Icon name="search" size={20} />
        <TextInput 
          placeholder="Search for a friend or partners" 
          style={styles.searchInput} 
          onTouchStart={() => navigation.navigate('SearchResult')}
        />
      </View>

      <Text style={styles.recommendText}>Recommend for You</Text>
      
      <View style={styles.categories}>
        {['All', 'Photography', 'Outdoor', 'Anime'].map(category => (
          <TouchableOpacity key={category} style={styles.categoryButton}>
            <Text>{category}</Text>
          </TouchableOpacity>
        ))}
      </View>

      <TouchableOpacity style={styles.recommendationCard} onPress={() => navigation.navigate('Profile')}>
        <Image source={require('../assets/snack-icon.png')} style={styles.recommendationImage} />
        <View style={styles.overlay}>
          <Text style={styles.overlayText}>⚡ Potential Match!</Text>
        </View>
        <View style={styles.recommendationInfo}>
          <Text style={styles.recommendationName}>Klee Gracia, 24</Text>
          <Text style={styles.recommendationBio}>Hi there! I'm Gracia ✨, a fun-loving adventurer looking for great connections.</Text>
        </View>
      </TouchableOpacity>

      <View style={styles.bottomNav}>
        {['Discover', 'Likes', 'Messages', 'Profile'].map(tab => (
          <TouchableOpacity 
            key={tab} 
            style={styles.bottomNavItem} 
            onPress={() => navigation.navigate(tab === 'Messages' ? 'Messages' : tab === 'Profile' ? 'Profile' : 'Homepage')}
          >
            <Icon name={tab === 'Discover' ? 'home' : tab.toLowerCase()} size={30} />
            <Text>{tab}</Text>
          </TouchableOpacity>
        ))}
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white'
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: 15
  },
  menuButton: {
    padding: 10
  },
  title: {
    fontSize: 18,
    fontWeight: 'bold'
  },
  headerIcons: {
    flexDirection: 'row'
  },
  stories: {
    flexDirection: 'row',
    paddingVertical: 10,
    paddingLeft: 15
  },
  story: {
    alignItems: 'center',
    marginRight: 15
  },
  storyImage: {
    width: 50,
    height: 50,
    borderRadius: 25
  },
  addStoryIcon: {
    position: 'absolute',
    bottom: 5,
    right: 0
  },
  searchBar: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#f1f1f1',
    borderRadius: 25,
    paddingHorizontal: 15,
    marginHorizontal: 15,
    marginVertical: 10
  },
  searchInput: {
    marginLeft: 10,
    flex: 1
  },
  recommendText: {
    fontSize: 16,
    fontWeight: 'bold',
    marginLeft: 15,
    marginTop: 10
  },
  categories: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginHorizontal: 15,
    marginVertical: 10
  },
  categoryButton: {
    borderRadius: 15,
    borderColor: 'gray',
    borderWidth: 1,
    paddingHorizontal: 15,
    paddingVertical: 5
  },
  recommendationCard: {
    margin: 15,
    padding: 15,
    backgroundColor: '#f1f1f1',
    borderRadius: 10,
    overflow: 'hidden'
  },
  recommendationImage: {
    width: '100%',
    height: 200
  },
  overlay: {
    position: 'absolute',
    top: 10,
    left: 10,
    backgroundColor: 'rgba(0,0,0,0.5)',
    borderRadius: 5,
    padding: 5
  },
  overlayText: {
    color: 'white'
  },
  recommendationInfo: {
    marginTop: 10
  },
  recommendationName: {
    fontSize: 16,
    fontWeight: 'bold'
  },
  recommendationBio: {
    fontSize: 14,
    color: 'gray'
  },
  bottomNav: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    paddingVertical: 10,
    borderTopWidth: 1,
    borderColor: '#f1f1f1'
  },
  bottomNavItem: {
    alignItems: 'center'
  }
});

export default HomepageScreen;

File: Scripts/results/generated_code_results/Flutter/Social Networking/SocialApp/HomepageScreen.dart
import 'package:flutter/material.dart';
class HomepageScreen extends StatefulWidget {
@override
_HomepageScreenState createState() => _HomepageScreenState();
}
class _HomepageScreenState extends State<HomepageScreen> {
int _currentIndex = 0;
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
leading: IconButton(
icon: Icon(Icons.menu),
onPressed: () {},
),
title: Text('Forging New Bonds and Friendships'),
actions: [
IconButton(
icon: Icon(Icons.notifications),
onPressed: () {},
),
IconButton(
icon: Icon(Icons.settings),
onPressed: () {},
),
],
),
body: Column(
children: [
Padding(
padding: EdgeInsets.all(8.0),
child: TextField(
decoration: InputDecoration(
hintText: 'Search for a friend or partners',
prefixIcon: Icon(Icons.search),
border: OutlineInputBorder(
borderRadius: BorderRadius.circular(20),
),
),
onTap: () {
Navigator.pushNamed(context, 'SearchResult');
},
),
),
Container(
height: 100,
child: ListView(
scrollDirection: Axis.horizontal,
children: [
_buildStoryAvatar('My Story', true),
_buildStoryAvatar('Hasley'),
_buildStoryAvatar('Javier'),
_buildStoryAvatar('Natalie'),
],
),
),
Row(
mainAxisAlignment: MainAxisAlignment.spaceEvenly,
children: [
TextButton(
child: Text('All'),
onPressed: () {},
),
OutlinedButton(
child: Text('Photography'),
onPressed: () {},
),
TextButton(
child: Text('Outdoor'),
onPressed: () {},
),
OutlinedButton(
child: Text('Anime'),
onPressed: () {},
),
],
),
Expanded(
child: ListView(
children: [
_buildProfileCard(),
],
),
),
],
),
bottomNavigationBar: BottomNavigationBar(
currentIndex: _currentIndex,
onTap: (index) {
setState(() {
_currentIndex = index;
});
if (index == 2) {
Navigator.pushNamed(context, 'Messages');
}
},
items: [
BottomNavigationBarItem(icon: Icon(Icons.explore), label: 'Discover'),
BottomNavigationBarItem(icon: Icon(Icons.favorite), label: 'Likes'),
BottomNavigationBarItem(icon: Icon(Icons.message), label: 'Messages'),
BottomNavigationBarItem(icon: Icon(Icons.person), label: 'Profile'),
],
),
);
}
Widget _buildStoryAvatar(String name, [bool isMyStory = false]) {
return Padding(
padding: EdgeInsets.symmetric(horizontal: 8.0),
child: Column(
children: [
CircleAvatar(
radius: 30,
backgroundImage: AssetImage('assets/images/template.png'),
child: isMyStory ? Icon(Icons.add, color: Colors.white) : null,
),
SizedBox(height: 4),
Text(name),
],
),
);
}
Widget _buildProfileCard() {
return Card(
margin: EdgeInsets.all(8.0),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Stack(
alignment: Alignment.topRight,
children: [
Image.asset('assets/images/template.png', fit: BoxFit.cover),
Container(
padding: EdgeInsets.all(4),
color: Colors.blue,
child: Text(
'Potential Match!',
style: TextStyle(color: Colors.white),
),
),
],
),
Padding(
padding: EdgeInsets.all(8.0),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text('Klee Gracia, 24', style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),
Text("Hi there! I'm Gracia ✨, a fun-loving adventurer looking for great connections."),
Wrap(
spacing: 8,
children: ['Running', 'Outdoor', 'Anime'].map((interest) =>
Chip(label: Text(interest))
).toList(),
),
],
),
),
ButtonBar(
children: [
TextButton(
child: Text('View Profile'),
onPressed: () {
Navigator.pushNamed(context, 'Profile');
},
),
ElevatedButton(
child: Text('Connect'),
onPressed: () {},
),
],
),
],
),
);
}
}